src/overlays/actors/ovl_En_Zl3/z_en_zl3.c

```diff
@@ -15,421 +19,2827 @@ void EnZl3_Destroy(Actor* thisx, GlobalContext* globalCtx);
+    s32 cond = Flags_GetSwitch(globalCtx, 0x37) &&
+               ((globalCtx->sceneNum == SCENE_GANON_DEMO) || (globalCtx->sceneNum == SCENE_GANON_FINAL) ||
+                (globalCtx->sceneNum == SCENE_GANON_SONOGO) || (globalCtx->sceneNum == SCENE_GANONTIKA_SONOGO));
+
+    if (cond) {
+        u8 curSpawn = globalCtx->curSpawn;
+
+        if ((func_80B54DB4(this) == 0x20) && (curSpawn == 0) && (gSaveContext.timer2Value <= 0)) {
+            return 1;
+        }
+    }
+    return 0;
+}
+
+void func_80B59828(EnZl3* this, GlobalContext* globalCtx) {
+    if (func_80B59698(this, globalCtx) || (!func_80B56EE4(this, globalCtx) && func_80B57890(this, globalCtx))) {
+        s16 newRotY;
+        
+        func_80B54E14(this, &D_06009FBC, 0, 0.0f, 0);
+        this->actor.flags |= 9;
+        func_80B56F10(this, globalCtx);
+        newRotY = func_80B571A8(this);
+        this->actor.shape.rot.y = newRotY;
+        this->actor.posRot.rot.y = newRotY;
+        this->unk_3C4 = this->actor.posRot.rot.z;
+        this->actor.shape.rot.z = 0;
+        this->actor.posRot.rot.z = this->actor.shape.rot.z;
+        this->action = 28;
+        this->drawConfig = 1;
+    } else {
+        Actor_Kill(&this->actor);
+    }
+
+    if (func_80B59698(this, globalCtx) != 0) {
+        func_80088AA0(180);
+        func_80B53468();
+        gSaveContext.healthAccumulator = 320;
+        Magic_Fill(globalCtx);
+        if (Flags_GetSwitch(globalCtx, 0x20)) {
+            Flags_UnsetSwitch(globalCtx, 0x20);
+            Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_BG_ZG, -144.0f, 3544.0f, -43.0f, 0, 0x2000, 0, 0x2000);
+        }
+        Flags_UnsetSwitch(globalCtx, 0x21);
+        Flags_UnsetSwitch(globalCtx, 0x22);
+        Flags_UnsetSwitch(globalCtx, 0x23);
+        Flags_UnsetSwitch(globalCtx, 0x24);
+        Flags_UnsetSwitch(globalCtx, 0x25);
+        Flags_UnsetSwitch(globalCtx, 0x26);
+        Flags_UnsetSwitch(globalCtx, 0x27);
+        Flags_UnsetSwitch(globalCtx, 0x28);
+        Flags_UnsetSwitch(globalCtx, 0x29);
+        Flags_UnsetSwitch(globalCtx, 0x2A);
+    }
+
+    if (func_80B54DB4(this) == 0x20) {
+        s32 cond;
+
+        func_80B54EA4(this, globalCtx);
+        cond = Flags_GetSwitch(globalCtx, 0x37) &&
+               ((globalCtx->sceneNum == SCENE_GANON_DEMO) || (globalCtx->sceneNum == SCENE_GANON_FINAL) ||
+                (globalCtx->sceneNum == SCENE_GANON_SONOGO) || (globalCtx->sceneNum == SCENE_GANONTIKA_SONOGO));
+        if (cond) {
+            func_80B53614(this, globalCtx);
+        }
+    }
+}
+
+void func_80B59A80(EnZl3* this, GlobalContext* globalCtx) {
+    if (func_80B59768(this, globalCtx)) {
+        Audio_PlaySoundGeneral(NA_SE_OC_REVENGE, &D_801333D4, 4, &D_801333E0, &D_801333E0, &D_801333E8);
+    }
+}
+
+void func_80B59AD0(EnZl3* this, GlobalContext* globalCtx) {
+    // todo look into
+    Actor* thisx = &this->actor; // unused, necessary to use 'this' first to fix regalloc
```

Comment:
Might be worth looking into if we can

Reply:
`if (thisx)` works too, but it seems worse. Also, strangely enough, this is the one function left that needs to be forward declared.